tests/suites/test_suite_pk.function

```diff
@@ -119,6 +119,7 @@ mbedtls_svc_key_id_t pk_psa_genkey( void )
     PSA_ASSERT( psa_generate_key( &attributes, &key ) );
 
 exit:
+    psa_reset_key_attributes( &attributes );
```

Comment:
> But the interface permits not calling `psa_reset_key_attributes` if you've only used `psa_set_key_xxx` (excluding domain parameters) and not `psa_get_key_attributes`. So we should test that there's no memory leak in such cases.

I know that and I think I tried to explain in the commit message why any way I prefer to do it systematically for educational purposes. With this approach, I think that the probability that a user not reading the interface documentation (what !) does not call `psa_reset_key_attributes` when he must is smaller. The probability that he does call it when it is not necessary is greater but has it doesn't hurt.

Reply:
My point is that if we systematically call `psa_reset_key_attributes` in tests, but there's a bug in the library that causes one of the functions like `psa_set_key_type` to allocate resources, we won't catch it.

Reply:
Ah, right. Then what about to not reset in tests focusing on `psa_set_xxx` APIs testing ?

Reply:
Yes, at least those. But more generally I think reset() shouldn't be called in functions that don't set domain parameters and don't call get_attributes().

Reply:
I cannot say I don't understand this point of view but the fact is that even in the library tests thus in code that is written by people knowing reasonably well PSA APIs some necessary calls are missing. Thus, this was an attempt to do something about it. I think that any alloc/free API couple is quite systematically a source of lots of bugs and here I would say it is even worst as from the name `psa_get_key_attributes` it is not completely obvious that this function does some allocations. Thus still in the spirit of trying to improve things but being less invasive, I could limit what I have done to the test functions where it is necessary to call `psa_reset_key_attributes` and also add some comment with it. That way the reset call  would gain some visibility. What do you think?

Reply:
We could instrument `psa_get_key_attributes` to allocate domain parameters under `MBEDTLS_TEST_HOOKS`. This way Asan would complain if we forgot reset(). But this is all out of scope here.

For this PR, I'd prefer to only add `psa_reset_key_attributes` in new cases and optionally in cases where it was necessary but forgotten before, but not to add it in places where it was definitely optional, and not to do any other related work.

Reply:
ok I wonder how to do that. I revert the commit adding the reset and add on top a new commit adding just a few resets. Then at the end when the PR is approved I rewrite the history removing the "reset commit" and its revert commit with an identical end result? Another possibility is to get rid of the reset commit now through a force-push but it may be disturbing for the end of the review. Let me know what you prefer.

Reply:
I think it would be best to wait until Bence has finished reviewing, then modify the commit.